Frends.AmazonS3.DownloadObject/Frends.AmazonS3.DownloadObject.Tests/UnitTest1.cs

```diff
@@ -0,0 +1,554 @@
+        {
+            _connection = new Connection()
+            {
+                AuthenticationMethod = AuthenticationMethod.AWSCredentials,
+                AwsAccessKeyId = _accessKey,
+                AwsSecretAccessKey = _secretAccessKey,
+                BucketName = _bucketName,
+                Region = Region.EuCentral1,
+            };
+
+            _input = new Input()
+            {
+                DestinationPath = @$"{_dir}\Download",
+                S3Directory = "2022/",
+                SearchPattern = "*"
+            };
+
+            _options = new Options()
+            {
+                DeleteSourceFile = true,
+                DownloadFromCurrentDirectoryOnly = true,
+                Overwrite = true,
+                ThrowErrorIfNoMatch = false,
+            };
+
+            var result = AmazonS3.DownloadObject(_connection, _input, _options, default);
+            Assert.IsNotNull(result.Result.Results);
+            Assert.IsTrue(File.Exists(@$"{_dir}\Download\Testfile.txt"));
+            Assert.IsTrue(File.Exists(@$"{_dir}\Download\Overwrite.txt"));
+            Assert.IsFalse(File.Exists(@$"{_dir}\Download\DownloadFromCurrentDirectoryOnly.txt"));
+        }
+
+        #endregion aws creds
+
+
+        /// <summary>
+        /// Create pre-signed URL.
+        /// </summary>
+        private string CreatePreSignedURL(string key)
+        {
+            AmazonS3Client? client = new(_accessKey, _secretAccessKey, RegionEndpoint.EUCentral1);
+
+            GetPreSignedUrlRequest request = new()
+            {
+                BucketName = _bucketName,
+                Key = key,
+                Verb = HttpVerb.GET,
+                Expires = DateTime.UtcNow.AddMinutes(2)
+            };
+            var url = client.GetPreSignedURL(request);
+            return url;
+        }
+
+
+        private void CreateTestFiles()
+        {
+            Directory.CreateDirectory($@"{_dir}\Download");
+            Directory.CreateDirectory($@"{_dir}\DownloadTestFiles\DownloadFromCurrentDirectoryOnly");
+            File.AppendAllText($@"{_dir}\Download\Overwrite.txt", "To Be Overwriten");
+            File.AppendAllText($@"{_dir}\DownloadTestFiles\Overwrite.txt", $"Overwrite complete {DateTime.UtcNow}");
+            File.AppendAllText($@"{_dir}\DownloadTestFiles\Testfile.txt", $"Test {DateTime.UtcNow}");
+            File.AppendAllText($@"{_dir}\DownloadTestFiles\DownloadFromCurrentDirectoryOnly\DownloadFromCurrentDirectoryOnly.txt", $"This should exists if DownloadFromCurrentDirectoryOnly = true.  {DateTime.UtcNow}");
+        }
+
+        private bool CompareFiles()
+        {
+            string mainFile = File.ReadAllText($@"{_dir}\Download\Overwrite.txt");
+            return mainFile.Contains("Overwrite complete") && !mainFile.Contains("To Be Overwriten") ? true : false;
+        }
+
+        /// <summary>
+        /// Delete test files and folders.
+        /// </summary>
+        [TestCleanup]
```

Comment:
Cleanup should be moved under the setup. Also, you can use `OneTimeTearDown`, since you only have to delete the directory after all tests have run, not before each test.

Reply:
Some of these files are overwritten by tests so it was easier to delete and create again to make sure they ready for the next test, and also needed less duplicate code.